docs/proposals/20201125-jaeger-traces-store.md

```diff
@@ -0,0 +1,161 @@
+
+#### Single Jaeger instance - soft multitenancy
+
+Soft multitenancy in Jaeger is not supported at the moment. The progress is tracked in [jaegertracing/jaeger/3427](https://github.com/jaegertracing/jaeger/issues/3427).
+
+Here are few issues that needs to be addressed. The list might not be complete:
+* Service and operation name API. The service query API nor storage does not expose/store labels.
+* Dependency/service architecture diagram. To support this feature dependency schema, query, collector and Spark aggregation would have to change.
+* Data retention/TTL configurable pre tenant. The data retention in Elasticsearch is configurable per index.
+
+### Jaeger Query
+
+There are two ways users could access data stored in Jaeger:
+1. Expose only Jaeger Query API in the Observatorium API.
+2. Expose Jaeger UI outside the cluster.
+
+#### Expose only Jaeger Query API
+
+Jaeger query API would be exposed in the Observatorium API and users would have to deploy Jaeger UI and use Observatorium API as a data source.
+
+Jaeger query exposes multiple HTTP and gRPC APIs. The Jaeger UI uses HTTP API. This API does not have OpenAPI schema, and it is only defined in [http_handler.go](https://github.com/jaegertracing/jaeger/blob/0dd3e2da0579caed9e24ad2782a1f638ad63214d/cmd/query/app/http_handler.go#L119). The API uses this [data model](https://github.com/jaegertracing/jaeger/blob/9cd7a7ec1aa43b24a8a970eb5b393ca2ffd98a5d/model/json/model.go#L52).
+
+#### Expose Jaeger UI
+
+In addition to exposing Jaeger query API the Observatorium could deploy Jaeger UI as well. At the moment Observatorium does not expose any UI and it is expected that users deploy UI locally.
+
+## Alternatives
+
+### Grafana Tempo
+
+[Grafana Tempo](https://github.com/grafana/tempo) is an alternative storage that Observatorium project could use to store traces. The project is inspired and similar to the Prometheus TSDB, Thanos and Grafana Loki, therefore could it could be a good fit for Obeservatorium. The storage is split into two parts: 1. low retention requiring fast local disk (e.g. SSD) to store recent data and 2. object store (S3 compatible) to store historical data. Tempo is a distributed system with [multiple components](https://grafana.com/docs/tempo/latest/operations/architecture/): distributor, ingester, compactor, storage, querier. Distributor uses parts of the OpenTelemetry collector codebase to receive data in multiple formats (OTLP, Jaeger, Zipkin).
+
+The project at the moment supports [search](https://grafana.com/docs/tempo/latest/getting-started/tempo-in-grafana/#tempo-search) only for recent data held in memory with retention about 15 minutes, therefore it is not suitable as a storage for Jaeger that exposes richer search API that supports search by attributes, time range and service name.
```

Comment:
I glanced over the Tempo documentation and releases, it looks like they are very close to supporting full back-end search (see https://github.com/grafana/tempo/pull/1174), looks like this is slated for the `1.3` release which is next. So this might soon not be a concern at all?

Reply:
The proposal is based on what our team currently supports as part of our product. If we will in the future support tempo we will make sure it is used in Observatorium as well.

Reply:
👍🏽

Reply:
I would add something along the lines: 

> 
```suggestion
The project at the moment supports [search](https://grafana.com/docs/tempo/latest/getting-started/tempo-in-grafana/#tempo-search) only for recent data held in memory with retention about 15 minutes, therefore it is not suitable as a storage for Jaeger that exposes richer search API that supports search by attributes, time range and service name. NOTE: We might need to revisit this option once [search feature](grafana/tempo#1174) is production ready.

On top of the above, the current tracing team proficiency is around Elastic and Jaeger development. Given that it makes sense to postpone this alternative and revisit it later once there are more arguments supporting this case.
```